sklearn/tests/test_calibration.py

```diff
@@ -0,0 +1,180 @@
+            pc_clf.fit(this_X_train, y_train + 1, sample_weight=sw_train)
+            prob_pos_pc_clf_relabeled = pc_clf.predict_proba(this_X_test)[:, 1]
+            assert_array_almost_equal(prob_pos_pc_clf,
+                                      prob_pos_pc_clf_relabeled)
+
+            # Check invariance against relabeling [0, 1] -> [-1, 1]
+            pc_clf.fit(this_X_train, 2 * y_train - 1, sample_weight=sw_train)
+            prob_pos_pc_clf_relabeled = pc_clf.predict_proba(this_X_test)[:, 1]
+            assert_array_almost_equal(prob_pos_pc_clf,
+                                      prob_pos_pc_clf_relabeled)
+
+            # Check invariance against relabeling [0, 1] -> [1, 0]
+            pc_clf.fit(this_X_train, (y_train + 1) % 2,
+                       sample_weight=sw_train)
+            prob_pos_pc_clf_relabeled = \
+                pc_clf.predict_proba(this_X_test)[:, 1]
+            if method == "sigmoid":
+                assert_array_almost_equal(prob_pos_pc_clf,
+                                          1 - prob_pos_pc_clf_relabeled)
+            else:
+                # Isotonic calibration is not invariant against relabeling
+                # but should improve in both cases
+                assert_true(brier_score_loss(y_test, prob_pos_clf) >
+                            brier_score_loss((y_test + 1) % 2,
+                                             prob_pos_pc_clf_relabeled))
+
+
+def test_calibration_multiclass():
+    """Test calibration for multiclass """
+    # test multi-class setting with classifier that implements
+    # only decision function
+    clf = LinearSVC()
+    X, y = make_blobs(n_samples=100, n_features=2, random_state=42,
+                      cluster_std=3.0)
+    X_train, y_train = X[::2], y[::2]
+    X_test, y_test = X[1::2], y[1::2]
+    for method in ['isotonic', 'sigmoid']:
+        cal_clf = CalibratedClassifierCV(clf, method=method, cv=2)
+        cal_clf.fit(X_train, y_train)
+        probas = cal_clf.predict_proba(X_test)
+        assert_array_almost_equal(np.sum(probas, axis=1), np.ones(len(X_test)))
+        assert_true(clf.fit(X_train, y_train).score(X_test, y_test) <=
+                    cal_clf.fit(X_train, y_train).score(X_test, y_test))
+
+    # Test that calibration of a multiclass classifier decreases log-loss
+    X, y = make_blobs(n_samples=100, n_features=2, random_state=42,
+                      cluster_std=3.0)
+    X_train, y_train = X[::2], y[::2]
+    X_test, y_test = X[1::2], y[1::2]
+
+    clf = RandomForestClassifier(n_estimators=10, random_state=42)
+    clf.fit(X_train, y_train)
+    clf_probs = clf.predict_proba(X_test)
+    score = log_loss(y_test, clf_probs)
+
+    for method in ['isotonic', 'sigmoid']:
+        cal_clf = CalibratedClassifierCV(clf, method=method, cv=3)
+        cal_clf.fit(X_train, y_train)
+        cal_clf_probs = cal_clf.predict_proba(X_test)
+        cal_score = log_loss(y_test, cal_clf_probs)
+        assert_true(cal_score < score)
```

Comment:
It would be interesting to test the `log_loss` of a calibrated `LinearSVC` vs the `log_loss` of a naive scaling of the `decision_function` to the `[0, 1]` range.